litevideo/csc/rgb2rgb16f.py

```diff
@@ -0,0 +1,93 @@
+# rgb2rgb16f
+
+from litex.gen import *
+from litex.soc.interconnect.stream import *
+
+from litevideo.csc.common import *
+
+@CEInserter()
+class PIX2PIXFDatapath(Module):
+    """ 
+    Converts a 8 bit unsigned int represented by a pixel in 
+    the range [0-255] to a 16 bit half precision floating point 
+    pix_number defined in the range [0-1] 
+    """
+    latency = 2
+
+    def __init__(self, pix_w, pixf_w):
+        self.sink = sink = Record(pix_layout(pix_w))
+        self.source = source = Record(pixf_layout(pixf_w))
+
+        # # #
+
+        # delay pix signal
+        pix_delayed = [sink]
+        for i in range(self.latency):
+            pix_n = Record(pix_layout(pix_w))
+            self.sync += getattr(pix_n, "pix").eq(getattr(pix_delayed[-1], "pix"))
+            pix_delayed.append(pix_n)
+
+        # Hardware implementation:
+
+        # Stage 1
+        # Leading one detector
+
+        lshift = Signal(4)
+        frac_val = Signal(10)
+
+        self.sync += [
+            # Leading one detector
+            If( sink.pix[7]==1,
```

Comment:
Instead of listing all conditions, it's also probably possible to describe that in a more generic way.